### PR TITLE
protocol sdk - make MintAPIClient a class, with overridable http methods.  add full fledged mint examples.  refactor MintClient

### PR DESCRIPTION
**Pull request by oveddan**
Changes to `MintApliClient` and `MintClient`, with descriptions described in the [changeset](https://github.com/ourzora/zora-protocol/pull/368/files#diff-a82688f719217e381d1004eb7e6a401697b8c7be48afc8d7b3e261c2b7a3e6a3)

### CODE REVIEW

**Comment by oveddan** [comment]
Current dependencies on/for this PR:
* main
  * **PR #368** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/368?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ourzora/zora-protocol/368?utm_source=stack-comment).

**Comment by oveddan** [comment]
running into this; doesn't play well with Viem

Screenshot 2023-11-16 at 2 01 43 PM

File: packages/protocol-sdk/src/mint/mint-api-client.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
-import { retries, get, post } from "../apis/http-api-base";
+import * as httpClientBase from "../apis/http-api-base";
```
Comment (iainnash) [inline packages/protocol-sdk/src/mint/mint-api-client.ts:1]: nit: not really a fan of import * unless it’s import * as .

**Review by iainnash — APPROVED** [review]
lgtm

good to merge after re-running `prettier`.

**Pull request merged**